subsys/net/lib/nrf_cloud/coap/src/nrf_cloud_coap_transport.c

```diff
@@ -312,17 +346,26 @@ static void client_callback(int16_t result_code, size_t offset, const uint8_t *p
 	}
 	if (result_code == COAP_RESPONSE_CODE_UNAUTHORIZED) {
 		LOG_ERR("Device not authenticated; reconnection required.");
-		xfer->nrfc_cc->authenticated = false;
+		if (xfer) {
```

Comment:
```suggestion
		if (xfer->nrfc_cc) {
```
`user_data` won't be NULL, so `xfer` won't be NULL